lib/longleaf/events/register_event.rb

```diff
@@ -14,7 +14,8 @@ class RegisterEvent
     # @param file_rec [FileRecord] file record
     # @param app_manager [ApplicationConfigManager] the application configuration
     # @param force [boolean] if true, then already registered files will be re-registered
-    def initialize(file_rec:, app_manager:, force: false, checksums: nil)
+    # @param digest_provider [DigestProvider] object which provides digests for files being registered
```

Comment:
If you keep the digest providers separate, will you need a DigestProvider parent class?

Reply:
While the java developer in me really wants it to be an interface, my impression is that ruby practices encourage duck typing rather than an explicit parent class/interface. The implementations so far don't really have overlapping functionality, so I didn't end up making a parent class. That said, the `@param` stating it takes a `DigestProvider` doesn't make sense since that doesn't exist. It sounds like it should be `[#get_digests]` to say it has to respond to that, I'll update that